Helper/Serverside.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Copyright © PureClarity. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+namespace Pureclarity\Core\Helper;
+
+use Psr\Log\LoggerInterface;
+use Pureclarity\Core\Helper\Service\Url;
+use Magento\Framework\HTTP\Client\Curl;
+use Pureclarity\Core\Helper\Data as CoreHelper;
+use Pureclarity\Core\Model\Serverside\Request\Data;
+
+/**
+ * Helper class for core service functions.
+ */
+class Serverside
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var CoreHelper */
+    private $coreHelper;
+
+    /** @var Url */
+    private $serviceUrl;
+
+    /** @var Curl */
+    private $curl;
+
+    /** @var Serializer */
+    private $serializer;
+
+    /** @var bool */
+    private $dispatched = false;
+
+    /** @var mixed[] */
+    private $result;
+
+    /** @var bool */
+    private $isAdmin = false;
+
+    /** @var string */
+    private $storeId;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param CoreHelper $coreHelper
+     * @param Url $serviceUrl
+     * @param Curl $curl
+     * @param Serializer $serializer
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        CoreHelper $coreHelper,
+        Url $serviceUrl,
+        Curl $curl,
+        Serializer $serializer
+    ) {
+        $this->logger       = $logger;
+        $this->coreHelper   = $coreHelper;
+        $this->serviceUrl   = $serviceUrl;
+        $this->curl         = $curl;
+        $this->serializer   = $serializer;
+    }
+
+    /**
+     * @param bool $isAdmin
+     */
+    public function setIsAdmin($isAdmin)
+    {
+        $this->isAdmin = $isAdmin;
+    }
+
+    /**
+     * @param string $storeId
+     */
+    public function setStoreId($storeId)
+    {
+        $this->storeId = $storeId;
+    }
+
+    /**
+     * Builds the request array and executes the request
+     *
+     * @param Data $request
+     */
+    public function dispatch($request)
+    {
+        if ($this->dispatched) {
+            return;
+        }
+
+        $this->dispatched = true;
+
+        $requestBody = [
+            'appId' => $request->getAppId(),
+            'secretKey' => $request->getSecretKey(),
+            'events' => $request->getEvents(),
+            'zones' => $request->getZones()
+        ];
+
+        if ($this->isAdmin) {
+            $requestBody['currentUrl'] = 'magento-admin';
+        } else {
+            $requestBody['referer'] = $request->getReferer();
+            $requestBody['currentUrl'] = $request->getCurrentUrl() ?: 'magento-backend';
+            $requestBody['currency'] = $request->getCurrency();
+            $requestBody['visitorId'] = $request->getVisitorId();
+            $requestBody['sessionId'] = $request->getSessionId();
+        }
+
+        $requestBody['userAgent'] = $request->getUserAgent() ?: 'magento-backend';
```

Comment:
just a thought - have you checked what Capture makes of these user agents? Just need to confirm they are not treating as a bot request

Reply:
debugged Capture, as far as i can tell, it doesnt treat "magento-backend" userAgent as a bot (it was put in as the afterPlace order plugin could be called via a post action from the payment gateway, and it needed a value)